core/Types.h

```diff
@@ -312,6 +331,24 @@ TYPE(ClassType) {
 };
 CheckSize(ClassType, 8, 8);
 
+template <> inline TypePtr make_type<ClassType, core::SymbolRef &>(core::SymbolRef &ref) {
+    return TypePtr(TypePtr::Tag::ClassType, ref.rawId(), 0);
+}
+
+template <> inline TypePtr make_type<ClassType, core::SymbolRef>(core::SymbolRef &&ref) {
+    return TypePtr(TypePtr::Tag::ClassType, ref.rawId(), 0);
+}
```

Comment:
Just golfing on memory layout, but: what if you store the rawId in `value` instead? It's not used for anything at this point, and if the tags for inlined values always fit in the first three bits of the ptr, you could do a pretty quick check and then return `value`, avoiding the shift. You could also use the first three bits being non-null for the inlined check this way.

Reply:
I can explore that as a follow-up change and see if there's any perf change. I suspect that, with everything being in cache, there's no difference.